domain/annotation/state/state.go

```diff
@@ -128,16 +126,31 @@ func (st *State) getAnnotationsForModel(ctx context.Context, id annotations.ID,
 // This is separate from the getAnnotationsForModel because for non-model ID
 // Kinds we need to find the UUID of the ID before we retrieve annotations from
 // the corresponding annotation table.
-func (st *State) getAnnotationsForID(ctx context.Context, id annotations.ID, getAnnotationsStmt *sqlair.Statement, annotationUUIDParam annotationUUID) (map[string]string, error) {
+func (st *State) getAnnotationsForID(ctx context.Context, id annotations.ID) (map[string]string, error) {
 	db, err := st.DB()
 	if err != nil {
 		return nil, errors.Capture(err)
 	}
 
+	kindName, err := kindNameFromID(id)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+	getAnnotationsQuery := fmt.Sprintf(`
+SELECT (key, value) AS (&Annotation.*)
+FROM annotation_%s
```

Comment:
I've switched it up so that it gets the table name from a method. I am also really not a fan of this pattern of using `Sprintf` to build the query. To be honest I think its generally better to do the first suggestion, but that would have required more extensive changes and I don't want to get too bogged down in this PR which primarily has the intent of fixing the future error found in https://github.com/juju/juju/pull/18554